collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -49,14 +44,20 @@
 
     private static final Pattern snakeCasePattern = Pattern.compile("([a-z0-9])([A-Z])");
 
-    public JmxCollector(Reader in) throws IOException, ParseException, MalformedObjectNameException {
-        this((JSONObject)new JSONParser().parse(in));
+    public JmxCollector(Reader in) throws IOException, MalformedObjectNameException {
+        this((Map<String, Object>)new Yaml().load(in));
     }
-    public JmxCollector(String jsonConfig) throws ParseException, MalformedObjectNameException {
-        this((JSONObject)new JSONParser().parse(jsonConfig));
+    public JmxCollector(String yamlConfig) throws MalformedObjectNameException {
+        this((Map<String, Object>)new Yaml().load(yamlConfig));
     }
+    private JmxCollector(Map<String, Object> config) throws MalformedObjectNameException {
+        if(config == null) {  //Yaml config empty, set all to default values
```

Comment:
From a maintenance standpoint, it may be better to replace config with an empty map here. That way the defaults don't have to be kept up to date in two places.